### DISCUSSION TITLE
How to build abseil-cpp into shared libraries, rather that static ones?

### DISCUSSION THREAD

**Discussion by jiapei100** [abseil/abseil-cpp#1474] [Q&A]
Hi, all:

How to build abseil-cpp into shared libraries, rather that static ones???

It looks inside [CMakeLists.txt](https://github.com/abseil/abseil-cpp/blob/master/CMakeLists.txt), there is **NOWHERE** for me to select 
- shared
- static
??

No wonder **ALL** libraries built out currently are **.a** ones. There is **NO** **.so** libraries??

My question: how to simply build out all shared libraries rather that static ones, for **abseil** ?

Cheers

**Comment by derekmauro** [comment]
Use the global CMake flag [BUILD_SHARED_LIBS](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html).
